### PR TITLE
[tests-only][full-ci] add test coverage to open text file with /app/open-with-web endpoint

### PR DESCRIPTION
**Pull request by anon-pradip**
## Description

Added test to open text file with /app/open-with-web endpoint with and without app_name parameter.

## Related Issue

- Related to: https://github.com/owncloud/ocis/issues/9682

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 1]
```diff
@@ -171,3 +171,77 @@ Feature: collaboration (wopi)
...
+      }
+      """
+
+    Examples:
+      | app-endpoint                                              |
+      | /app/open-with-web?file_id=<<FILEID>>&app_name=FakeOffice |
+      | /app/open-with-web?file_id=<<FILEID>>                     |
+
+
+  Scenario: open text file in web with app name in url query (MIME type not registered in app-registry)
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:203]:
```suggestion
  Scenario: open text file using open-with-web with app name in url query (MIME type not registered in app-registry)
```

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 2]
```diff
@@ -171,3 +171,77 @@ Feature: collaboration (wopi)
...
+          "uri": {
+            "type": "string",
+             "pattern": "^.*?contextRouteName=files-spaces-personal&fileId=%file_uuidv4_pattern%%24%file_uuidv4_pattern%%21%file_uuidv4_pattern%$"
+          }
+        }
+      }
+      """
+
+  @issue-9928
+  Scenario: open text file in web without app name in url query (MIME type not registered in app-registry)
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:225]:
```suggestion
  Scenario: open text file using open-with-web without app name in url query (MIME type not registered in app-registry)
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -2317,6 +2317,14 @@ public function substituteInLineCodes(
...
 				],
 				"parameter" => []
+			],
+			[
+				"code" => "%file_uuidv4_pattern%",
+				"function" => [
+					__NAMESPACE__ . '\TestHelpers\GraphHelper',
+					"getUUIDv4Regex"
+				],
+				"parameter" => []
```
Comment (SagarGi) [inline tests/acceptance/bootstrap/FeatureContext.php:2327]: won't `file_id_pattern` regex work here or its different?

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 4]
```diff
@@ -171,3 +171,77 @@ Feature: collaboration (wopi)
       | app-endpoint                                     |
       | /app/open?file_id=<<FILEID>>&app_name=FakeOffice |
       | /app/open?file_id=<<FILEID>>                     |
+
+
+  Scenario Outline: open file with .odt extension in web
```
Comment (SagarGi) [inline tests/acceptance/features/apiCollaboration/wopi.feature:176]:
```suggestion
  Scenario Outline: open file with .odt extension (open-with-web)
```
may be this?

File: tests/acceptance/bootstrap/FeatureContext.php [hunk 5]
```diff
@@ -2317,6 +2317,14 @@ public function substituteInLineCodes(
 					"getUUIDv4Regex"
 				],
 				"parameter" => []
+			],
+			[
+				"code" => "%file_uuidv4_pattern%",
```
Comment (PrajwolAmatya) [inline tests/acceptance/bootstrap/FeatureContext.php:2322]:
```suggestion
				"code" => "%uuidv4_pattern%",
```

**Comment by saw-jan** [comment]
Wait for PR #9931

File: tests/acceptance/bootstrap/FeatureContext.php (hunk 3 above)
Comment (anon-pradip) [reply tests/acceptance/bootstrap/FeatureContext.php]: We have:
```
[
				"code" => "%file_id_pattern%",
				"function" => [
					__NAMESPACE__ . '\TestHelpers\GraphHelper',
					"getFileIdRegex"
				],
				"parameter" => []
			],
```
for `file_id_pattern` and the method `getFileIdRegex` used above returns `\$` as per:
```
public static function getFileIdRegex(): string {
		return self::getUUIDv4Regex() . '\\\$' . self::getUUIDv4Regex() . '!' . self::getUUIDv4Regex();
	}
```

but we only needed `$`.
So I think `file_id_pattern` won't work.

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Pull request merged**